src/Orchard.Cms.Web/Modules/Orchard.OpenId/OpenIddictConfigureOptions.cs

```diff
@@ -38,23 +39,57 @@ public void Configure(OpenIddictOptions openIddictOptions)
                 openIddictOptions.SigningCredentials.AddEphemeralKey();
                 openIddictOptions.AllowInsecureHttp = true;
             }
-            else
+            else if(openIdSettings.CertificateStoreLocation.HasValue && openIdSettings.CertificateStoreName.HasValue && !string.IsNullOrEmpty(openIdSettings.CertificateThumbPrint))
             {
-                openIddictOptions.AllowInsecureHttp = false;
-                if (openIdSettings.CertificateStoreLocation.HasValue && openIdSettings.CertificateStoreName.HasValue && !string.IsNullOrEmpty(openIdSettings.CertificateThumbPrint))
+                try
                 {
-                    try
-                    {
-
-                        openIddictOptions.SigningCredentials.AddCertificate(openIdSettings.CertificateThumbPrint, openIdSettings.CertificateStoreName.Value, openIdSettings.CertificateStoreLocation.Value);
-                    }
-                    catch (Exception e)
-                    {
-                        _logger.LogError("Orchard.OpenId module needs you provide a valid signing certificate.", e);
-                        throw e;
-                    }
+                    openIddictOptions.AllowInsecureHttp = false;
+                    openIddictOptions.SigningCredentials.Clear();
+                    openIddictOptions.SigningCredentials.AddCertificate(openIdSettings.CertificateThumbPrint, openIdSettings.CertificateStoreName.Value, openIdSettings.CertificateStoreLocation.Value);
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError("Orchard.OpenId module needs you provide a valid signing certificate.", e);
+                    throw e;
                 }
             }
+
+            if (openIdSettings.EnableAuthorizationEndpoint)
+            {
+                openIddictOptions.AuthorizationEndpointPath = "/Orchard.OpenId/Access/Authorize";
+            }
+            if (openIdSettings.EnableTokenEndpoint)
+            {
+                openIddictOptions.TokenEndpointPath = "/Orchard.OpenId/Access/Token";
+            }
+            if (openIdSettings.EnableLogoutEndpoint)
+            {
+                openIddictOptions.LogoutEndpointPath = "/Orchard.OpenId/Access/Logout";
+            }
+            if (openIdSettings.EnableUserInfoEndpoint)
+            {
+                openIddictOptions.UserinfoEndpointPath = "/Orchard.OpenId/Access/Userinfo";
+            }
+            if (openIdSettings.AllowPasswordFlow)
+            {
+                openIddictOptions.GrantTypes.Add(OpenIdConnectConstants.GrantTypes.Password);
+            }
+            if (openIdSettings.AllowClientCredentialsFlow)
+            {
+                openIddictOptions.GrantTypes.Add(OpenIdConnectConstants.GrantTypes.ClientCredentials);
+            }
+            if (openIdSettings.AllowAuthorizationCodeFlow || openIdSettings.AllowHybridFlow)
```

Comment:
Not sure I love this check, but if you really want to have an "enable hybrid flow" checkbox, I guess it's the only option.